### ISSUE TITLE
`odo describe component` should display actual resources created and not just their Devfile component names

### ISSUE THREAD

**Issue by valaparthvi** [redhat-developer/odo#6229]
/kind user-story

## User Story
As a user I want to `odo describe component` to show the k8s resources that are managed by odo and not just the name of the Devfile component So that I know what the resource really is.

## Acceptance Criteria 
- [ ] It should show k8s resource in the form of `<devfile component name>: <Resource>/<Resource Name>`

```sh
$ odo describe component
--- omitted for brevity ---

Container components:
 •  runtime : registry.access.redhat.com/ubi8/nodejs-14:latest

Kubernetes components:
 •  outerloop-deployment: Deployment/my-nodejs-app
 •  outerloop-service: Service/my-nodejs-app
 •  outerloop-url: Ingress/my-nodejs-app

```

## Links 
- Related Epic (mandatory): 

/kind user-story

## Reproduction steps
1. Use the Devfile defined in the advanced guide(https://odo.dev/docs/user-guides/advanced/deploy/nodejs/) to create a component.
2. `odo deploy` to deploy it on the cluster.
3. `odo describe component` and check the section on "Kubernetes Component"

**Comment by kadel** [comment]
I agree that the new suggested output is better, but this is not a bug.

It works as it was originally intended (https://github.com/redhat-developer/odo/issues/5661)

This is a feature request or feature enhancement.

It should be converted to a user story

**Comment by rm3l** [comment]
/remove-lifecycle rotten
/remove-lifecycle stale
/lifecycle frozen
/reopen

Still relevant.
